src/http/test/http_test.cpp

```diff
@@ -307,4 +309,161 @@ DOCTEST_TEST_CASE("Escaping")
     DOCTEST_CHECK(m.path == "/foo/bar");
     DOCTEST_CHECK(m.query == "this=that&awkward=escaped string :;-=?!\"");
   }
+
+  {
+    const std::string request =
+      "GET "
+      "/hello%20world?hello%20world=hello%20world&saluton%20mondo=saluton%"
+      "20mondo HTTP/1.1\r\n\r\n";
+
+    http::SimpleRequestProcessor sp;
+    http::RequestParser p(sp);
+
+    const std::vector<uint8_t> req(request.begin(), request.end());
+    auto parsed = p.execute(req.data(), req.size());
+
+    DOCTEST_CHECK(!sp.received.empty());
+    const auto& m = sp.received.front();
+    DOCTEST_CHECK(m.method == HTTP_GET);
+    DOCTEST_CHECK(m.path == "/hello%20world");
+    DOCTEST_CHECK(
+      m.query == "hello world=hello world&saluton mondo=saluton mondo");
+  }
+}
+
+struct SignedRequestProcessor : public http::SimpleRequestProcessor
+{
+  std::queue<ccf::SignedReq> signed_reqs;
+
+  virtual void handle_request(
+    http_method method,
+    const std::string_view& path,
+    const std::string& query,
+    http::HeaderMap&& headers,
+    std::vector<uint8_t>&& body) override
+  {
+    const auto signed_req = http::HttpSignatureVerifier::parse(
+      http_method_str(method), path, query, headers, body);
+    DOCTEST_REQUIRE(signed_req.has_value());
+
+    signed_reqs.push(signed_req.value());
+
+    http::SimpleRequestProcessor::handle_request(
+      method, path, query, std::move(headers), std::move(body));
+  }
+};
+
+DOCTEST_TEST_CASE("Signatures")
+{
+  // Produce signed requests with some formatting variations, ensure we can
+  // parse them
+  auto kp = tls::make_key_pair();
+
+  http::Request request("/foo", HTTP_POST);
+  request.set_query_param("param", "value");
+  request.set_query_param("pet", "dog");
+  request.set_header("Host", "example.com");
+  request.set_header("Date", "Sun, 05 Jan 2014 21:31:40 GMT");
+  request.set_header("Content-Type", "application/json");
```

Comment:
This is based on the test vectors from the draft RFC. I was going to use them directly, but our minimal crypto doesn't support the curve used with those test values, so instead we just test that we can parse our own produced values.